### PR TITLE
Added fix for dynamic sessionServiceName

### PR DESCRIPTION
**Pull request by bmeyers22**

### CODE REVIEW

File: lib/torii/routing/application-route-mixin.js [hunk 1]
```diff
@@ -11,7 +13,7 @@ export default Ember.Mixin.create({
     }
   },
   checkLogin: function () {
-    return this.session.fetch()
+    return this[configuration.sessionServiceName].fetch()
```
Comment (mixonic) [inline lib/torii/routing/application-route-mixin.js]: I'd prefer `this.get(configuration.sessionServiceName).fetch()` - just a little safer.

File: lib/torii/routing/authenticated-route-mixin.js [hunk 2]
```diff
@@ -11,11 +13,12 @@ export default Ember.Mixin.create({
     }
   },
   authenticate: function (transition) {
-    var route = this;
-    var isAuthenticated = this.get('session.isAuthenticated');
+    var route = this,
+      session = this[configuration.sessionServiceName],
```
Comment (mixonic) [inline lib/torii/routing/authenticated-route-mixin.js]: Again `this.get(configuration.sessionServiceName)`

File: lib/torii/routing/authenticated-route-mixin.js [hunk 3]
```diff
@@ -11,11 +13,12 @@ export default Ember.Mixin.create({
...
+      isAuthenticated = this.get(configuration.sessionServiceName + '.isAuthenticated');
```
Comment (mixonic) [inline lib/torii/routing/authenticated-route-mixin.js]: And then this can `session.get('isAuthenticated')`

**Comment by mixonic** [comment]
:+1: good catch @bmeyers22. I would love a test if you could add one (should be fairly easy to add).

**Comment by mwpastore** [comment]
@mixonic Won't the tests for the `authenticatedRoute()` DSL cover these changes?

**Comment by bmeyers22** [comment]
As far as tests, I believe the correct place would be the configuration tests, which should already cover that configs defined can be read properly from the module

**Comment by mixonic** [comment]
Seems like a basic test in https://github.com/Vestorly/torii/blob/master/test/tests/acceptance/routing-test.js would be appropriate, but with the session service name (like [here](https://github.com/Vestorly/torii/blob/master/test/tests/acceptance/routing-test.js#L21)) set to something other that `session`

**Comment by bmeyers22** [comment]
@mixonic I wrote a test, let me know if this was the kind of thing you were looking for

**Comment by mixonic** [comment]
@bmeyers22 perfect! Thank you. Can you rebase down to single squashed commit? I'd love to merge, I will push out beta.5 today and hopefully that is the last.

**Comment by bmeyers22** [comment]
All set

**Comment by mixonic** [comment]
:heart_eyes: thank you for the contribution!

**Comment by mixonic** [comment]
In 0.6.0-beta.7

**Pull request merged**